src/main/java/com/suse/salt/netapi/calls/LocalCall.java

```diff
@@ -143,33 +144,38 @@ public LocalCall(String functionName, Optional<List<?>> arg,
      */
     public Map<String, Result<R>> callSync(final SaltClient client, Target<?> target)
             throws SaltException {
-        Map<String, Object> customArgs = new HashMap<>();
-        customArgs.put("tgt", target.getTarget());
-        customArgs.put("expr_form", target.getType());
-
-        Type xor = parameterizedType(null, Result.class, getReturnType().getType());
-        Type map = parameterizedType(null, Map.class, String.class, xor);
-        Type listType = parameterizedType(null, List.class, map);
-        Type wrapperType = parameterizedType(null, Return.class, listType);
+        return callSyncHelper(client, target, Optional.empty(),
+                Optional.empty(), Optional.empty(), Optional.empty()).get(0);
+    }
 
-        @SuppressWarnings("unchecked")
-        Return<List<Map<String, Result<R>>>> wrapper = client.call(this,
-                Client.LOCAL, "/",
-                Optional.of(customArgs),
-                (TypeToken<Return<List<Map<String, Result<R>>>>>)
-                TypeToken.get(wrapperType));
-        return wrapper.getResult().get(0);
+    /**
+     * Calls a execution module function on the given target with batching and
+     * synchronously waits for the result. Authentication is done with the token
+     * therefore you have to login prior to using this function.
+     *
+     * @param client SaltClient instance
+     * @param target the target for the function
+     * @param batch  the batch specification
+     * @return A list of maps with each list representing each batch, and maps containing
+     * the results with the minion names as keys.
+     * @throws SaltException if anything goes wrong
+     */
+    public List<Map<String, Result<R>>> callSync(final SaltClient client, Target<?> target,
+            Batch batch)
+            throws SaltException {
+        return callSyncHelper(client, target, Optional.empty(),
+                Optional.empty(), Optional.empty(), Optional.of(batch));
     }
 
     /**
      * Calls a execution module function on the given target and synchronously
      * waits for the result. Authentication is done with the given credentials
      * no session token is created.
      *
-     * @param client SaltClient instance
-     * @param target the target for the function
-     * @param username username for authentication
-     * @param password password for authentication
+     * @param client     SaltClient instance
+     * @param target     the target for the function
+     * @param username   username for authentication
+     * @param password   password for authentication
```

Comment:
@briancecker: Do not open another PR to improve this patch. It is OK to force push this PR if you like, but from my perspective there is no need to end up with just a single commit. I personally prefer to just add more commits fixing up things, but do as you like.